addons/event_booth_sale/models/sale_order_line.py

```diff
@@ -94,10 +94,9 @@ def _get_display_price(self):
         if self.event_booth_pending_ids and self.event_id:
             company = self.event_id.company_id or self.env.company
             currency = company.currency_id
-            product = self.event_booth_pending_ids.booth_category_id.product_id
-            pricelist = product.product_tmpl_id._get_contextual_pricelist()
```

Comment:
I see that this shouldn't change anything to the end result, but it's not super clear why we remove it.

Maybe reference this commit: c2dbf302f34a8f3b3f6c344f6792c4795e41fc82
as it seems to back-port that change

Reply:
@thiryr yes, it fixes the issue. "_get_contextual_pricelist" return nothing when you are creating a SO line from the backend. It only returns a pricelist when it comes from the frontend. So we need to take the pricelist from the SO.

Then I removed 'product =....' as it is not used anymore

Reply:
I meant it's not clear why we wouldn't just add a fallback when pricelist is not set instead of fully replacing it, from the commit message. The previous on 16.2 specifies that it's actually just the same thing. Plus referencing it in the commit message also justifies merging it here more easily.

But fine by me either way (I don't have r+ though, fyi)

Reply:
ok, thanks for your review.